src/main/java/it/academy/app/services/ApplicationFormService.java

```diff
@@ -0,0 +1,131 @@
+        query.put("idHash", id);
+        BasicDBObject dbObject = (BasicDBObject) collection.findOne(query);
+        if (dbObject == null)
+            throw new IncorrectDataException("Incorrect id");
+        return setApplicationForm(dbObject);
+    }
+
+    //add test
+    public List<ApplicationForm> getAllApplications() {
+        DBCursor cursor = collection.find();
+        List<ApplicationForm> applicationForms = new ArrayList<>();
+        while (cursor.hasNext()) {
+            applicationForms.add(setApplicationForm((BasicDBObject) cursor.next()));
+        }
+        return applicationForms;
+    }
+
+    //add test
+    public ApplicationForm changeStatus(ApplicationForm applicationForm) throws IncorrectDataException {
+        ObjectId objectId = new ObjectId(applicationForm.getId());
+        StatusChangeValidator validator = new StatusChangeValidator();
+        validator.checkIsStatusInProgress(findApplicationFormById(objectId).getStatus());
+        BasicDBObject searchQuery = new BasicDBObject().append("_id", objectId);
+        BasicDBObject newStatus = new BasicDBObject().append("status", applicationForm.getStatus());
+        BasicDBObject newDocument = new BasicDBObject().append("$set", newStatus);
+        collection.update(searchQuery, newDocument);
+        ApplicationForm applicationFormWithNewStatus = findApplicationFormById(objectId);
+        if (applicationFormWithNewStatus.getStatus().equals("PATVIRTINTA")) {
+            emailService.sendMail(applicationFormWithNewStatus);
+        }
+        return applicationFormWithNewStatus;
+    }
+    //add test
+    public ApplicationForm createNewForm(ApplicationForm applicationForm) {
+        BasicDBObject formToAdd = new BasicDBObject();
+        formToAdd.put("email", applicationForm.getEmail());
+        formToAdd.put("education", applicationForm.getEducation());
+        formToAdd.put("name", applicationForm.getName());
+        formToAdd.put("surname", applicationForm.getSurname());
+        formToAdd.put("phone", applicationForm.getPhone());
+        formToAdd.put("freeTimeActivity", applicationForm.getFreeTimeActivity());
+        formToAdd.put("threePartyAgreement", applicationForm.getThreePartyAgreement());
+        formToAdd.put("available14To18", applicationForm.getAvailable14To18());
+        formToAdd.put("motivation", applicationForm.getMotivation());
+        formToAdd.put("experience", applicationForm.getExperience());
+        formToAdd.put("infoAboutAcademy", applicationForm.getInfoAboutAcademy());
+        formToAdd.put("status", "PERŽIŪRIMA");
+        String currentDateTime = dateFormat.format(Date.from(java.time.ZonedDateTime.now().toInstant()));
+        formToAdd.put("dateTime", currentDateTime);
+        String uniqueId = currentDateTime.concat(applicationForm.getEmail());
+        formToAdd.put("idHash", hashService.getHash(uniqueId));
+        collection.save(formToAdd);
+        emailService.sendMail(setApplicationForm(formToAdd));
```

Comment:
This call should probably not be inside createNewForm() method. Extract it after this method's call.
Same goes for setApplicationForm()